lib/ImportFromC/HeaderToModel.cpp

```diff
@@ -844,37 +814,25 @@ bool DeclVisitor::handleStructType(const clang::RecordDecl *RD) {
       return false;
     }
 
-    std::optional<model::Register::Values> LocationID;
+    model::Register::Values Location;
     if (AnalysisOption == ImportFromCOption::EditFunctionPrototype) {
-      if (not Field->hasAttr<AnnotateAttr>()) {
-        revng_log(Log,
-                  "Struct field representing return value should have annotate "
-                  "attribute describing location");
-        return false;
-      }
-
-      auto Annotate = std::find_if(begin(Field->getAttrs()),
-                                   end(Field->getAttrs()),
-                                   [&](Attr *Attribute) {
-                                     return isa<AnnotateAttr>(Attribute);
-                                   });
-
-      auto Loc = getLoc(cast<AnnotateAttr>(*Annotate)->getAnnotation());
-      if (not Loc or Loc->empty()) {
-        revng_log(Log,
-                  "Return value should have attribute annotate with reg or "
-                  "stack");
-        return false;
-      }
-
-      // TODO: Handle stack location.
-      if (*Loc == "stack") {
-        revng_log(Log, "We don't support Return value on stack for now");
-        return false;
+      std::optional Register = parseStringAnnotation(*Field, RegAnnotation);
+      if (not Register.has_value()) {
+        std::optional Stack = parseStringAnnotation(*Field, StackAnnotation);
+        if (Stack.has_value()) {
+          // TODO: Handle stack location.
+          revng_log(Log, "We don't support stack parameters in RFTs for now");
+          return false;
+        } else {
+          revng_log(Log,
+                    "Parameters must have either a register or a stack "
+                    "annotation.");
+          return false;
```

Comment:
Also I think here we're assuming this is the return struct type of a RFT. The same restriction on having just one stack return value which is a pointer should apply I think.

Reply:
Done! Please take a look at all the new tests :)

(I squashed them into the tests commit though)

Reply:
Cool thanks!